### ISSUE TITLE
Hi, I tried running scripts/train_avdd.sh, but the process gets killed shortly after it starts.

### ISSUE THREAD

**Issue by mortyzhou77** [sakshamsingh1/AVDD#4]
Is it possible that the training process is using too much memory？

Could you please let me know what CPU and how much RAM your system has when running train_avdd.sh?

Thank u so much！

**Comment by sakshamsingh1** [comment]
Hi, were you able to train with IPC=1 on the VGG_subset? It should be manageable on a 24GB GPU.

For higher IPC values and larger datasets (i.e. more classes), the memory requirements increase significantly.

Since distillation can be performed class-wise, I trained each class separately. 
I'm currently working on cleaning up the code and will release a script for parallel class-wise training soon.

**Comment by sakshamsingh1** [comment]
Got it. To speed up computation, the entire processed training data (ave_train.pt, ~17GB) is loaded into CPU memory. This might be causing the CPU to run OOM.

Meanwhile I am cleaning and will release the class-wise training scripts. Please try these steps: Split train_data by class -> minimize the loss separately for each class.

**Comment by sakshamsingh1** [comment]
Hi @Mortyzhou-Shef-BIT classwise training scripts have been released. Please let me know if you still have OOM issues.

**Comment by mortyzhou77** [comment]
> Hi [@Mortyzhou-Shef-BIT](https://github.com/Mortyzhou-Shef-BIT) classwise training scripts have been released. Please let me know if you still have OOM issues.

Thank u so much!  I will try it!
